docs/source/native/class.rst

```diff
@@ -0,0 +1,87 @@
+Classes
+=======
+
+
+
+.. note::
+
+   This is by no means complete. Please add anything you know.
+
+Declaring Classes
+-----------------
+
+To declare a class, first add the ``untyped`` keyword and the class as a global variable at file level.
+
+.. code-block:: javascript
+
+    untyped
+    global var ExampleClass
+
+You can then add the declaration inside of a function.
+
+Most classes use a constructor. A constructor is a function of the instance that gets executed on object creation.
+
+.. code-block:: javascript
+
+    void function initClient() {
+        class ExampleClass {
+            constructor(){print("Instance of ExampleClass created");}
+        }
+    }
+
+You can require parameters in the constructor. Keep in mind that you have to pass those when creating an object.
+
+The types of the parameters have to be ``var``, but you can ``expect`` types in the constructor as usual.
+
+Usually objects have properties. To define them, just add their identifier into the class without type declaration. The properties will be of type ``var``. However, you are required to set a default value of a property. This may be ``null``.
+
+Every object has a reference to itself called ``this``. You can change parameters of an object by reference.
+
+.. code-block:: javascript
+
+    void function initClient() {
+        class ExampleClass {
+            property = null
+            constructor( var parameter ) {
+                this.property = expect int(parameter);
+            }
+        }
+    }
+
+You can't use the class name as a type. Instead use ``var`` instead. You can't ``expect`` them either.
+
+To declare an object, do:
+
+.. code-block:: javascript
+
+    var exampleObject = ExampleClass(1);
+    int n = exampleObject.property // n = 1
```

Comment:
Does the field `property` need to be added by the constructor, or can it be implicitly generated at runtime? It hasn't been used before, so it's ambiguous here.

Reply:
It's possible to insert fields at runtime, but only if the class hasn't been instantiated yet.